datatypes/src/operations/reproject.rs

```diff
@@ -239,34 +240,29 @@ pub trait ReprojectClipped<P: CoordinateProjection> {
     fn reproject_clipped(&self, projector: &P) -> Result<Self::Out>;
 }
 
-impl<P> ReprojectClipped<P> for BoundingBox2D
+impl<P, A> ReprojectClipped<P> for A
 where
     P: CoordinateProjection,
+    A: AxisAlignedRectangle,
 {
-    type Out = BoundingBox2D;
-    fn reproject_clipped(&self, projector: &P) -> Result<BoundingBox2D> {
+    type Out = A;
+    fn reproject_clipped(&self, projector: &P) -> Result<A> {
         const POINTS_PER_LINE: i32 = 7;
 
         // clip bbox to the area of use of the target projection
         let area_of_use_projector = CoordinateProjector::from_known_srs(
             SpatialReference::epsg_4326(),
             projector.source_srs(),
         )?;
-        let source_area_of_use = projector.source_srs().area_of_use()?;
-        let target_area_of_use = projector.target_srs().area_of_use()?;
-        let area_of_use = source_area_of_use.intersection(&target_area_of_use).ok_or(
-            Error::BboxesDoNotIntersect {
-                bbox_a: source_area_of_use,
-                bbox_b: target_area_of_use,
-            },
-        )?;
+        let source_area_of_use = projector.source_srs().area_of_use::<A>()?;
+        let target_area_of_use = projector.target_srs().area_of_use::<A>()?;
+        let area_of_use = source_area_of_use
+            .intersection(&target_area_of_use)
+            .ok_or(Error::SpatialBoundsDoNotIntersect)?;
```

Comment:
i know there are no generic errors. maybe we can have coords in the error?

Reply:
cf my other comment